src/generate-types.ts

```diff
@@ -352,6 +352,26 @@ export function run(options: Options) {
     };
   }
 
+  function generateHeadersSchemaType(headers: oas.HeadersObject): oas.SchemaObject {
+    const required: string[] = [];
+    const properties = Object.entries(headers).reduce((memo, [headerName, headerObject]) => {
+      if (oautil.isReferenceObject(headerObject)) {
+        required.push(headerName);
+        return { ...memo, [headerName]: headerObject };
+      } else if (headerObject.schema) {
+        required.push(headerName);
+        return { ...memo, [headerName]: headerObject.schema };
+      }
+      return memo;
+    }, {} as { [key: string]: oas.SchemaObject | oas.ReferenceObject });
+    return {
+      type: 'object',
+      properties,
+      required,
```

Comment:
should we make the declared headers `required`? 🤔

Reply:
The specification doesn't mention it: https://swagger.io/docs/specification/describing-responses/

Reply:
There is a required flag you can give to the header parameter. If the thing is required then yes otherwise I think not.

https://swagger.io/docs/specification/describing-parameters/#header-parameters

Reply:
Hm, what about `$ref` headers? is there an easy way to "resolve" the ref so we can check if that's required?

Reply:
Point. not really as such if we consider that the ref can be to an another file. Best we could do is emit the equivalent of `undefined | parameterType` for parameter components, but it would still require the property to be there. Unless we want to have a look on the runtime type checking code for undefined values.

making $ref optional might be ok or even making everything optional

Reply:
Copoint. It actually might be that we inline parameter schemas. so this should actually work out of the box (mostly bc inlining would fail for out of file references)

yes. we do if you look eg at generateParameterType in generate-types.ts it uses deref which does this.